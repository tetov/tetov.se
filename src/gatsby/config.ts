import type { GatsbyConfig } from "gatsby"

import path from "path"

const config: GatsbyConfig = {
  siteMetadata: {
    title: `tetov xyz`,
    description: `Project and blog`,
    siteUrl: `https://tetov.xyz`,
  },
  plugins: [
    {
      resolve: `gatsby-plugin-feed`,
      options: {
        query: `
          {
            site {
              siteMetadata {
                title
                description
                siteUrl
                site_url: siteUrl
              }
            }
          }
        `,
        feeds: [
          {
            serialize: ({ query: { site, allMarkdownRemark } }) => {
              return allMarkdownRemark.nodes.map(node => {
                return Object.assign({}, node.frontmatter, {
                  description: node.excerpt,
                  date: node.frontmatter.date,
                  url: site.siteMetadata.siteUrl + node.fields.slug,
                  guid: site.siteMetadata.siteUrl + node.fields.slug,
                  custom_elements: [{ "content:encoded": node.html }],
                })
              })
            },
            query: `
              {
                allMarkdownRemark(
                  sort: { order: DESC, fields: [frontmatter___date] },
                ) {
                  nodes {
                    excerpt
                    html
                    fields {
                      slug
                    }
                    frontmatter {
                      title
                      date
                    }
                  }
                }
              }
            `,
            output: "/rss.xml",
          },
        ],
      },
    },
    `gatsby-plugin-image`,
        {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `tetov-xyz`,
        short_name: `tetov-xyz`,
        start_url: `/`,
        background_color: `#663399`,
        theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/site_icon.png`, // This path is relative to the root of the site.
      },
    },
    `gatsby-plugin-react-helmet`,
    {
      resolve: `gatsby-plugin-sharp`,
      options: {
        defaultQuality: 75,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `blog`,
        path: path.resolve(__dirname, "../../content/blog"),
      }
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: path.resolve(__dirname, "../images"),
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `projs`,
        path: path.resolve(__dirname, "../../content/projs"),
      }
    },
    {
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          {
            resolve: `gatsby-remark-images`,
            options: {
              maxWidth: 630,
            },
          },
          {
            resolve: `gatsby-remark-responsive-iframe`,
            options: {
              wrapperStyle: `margin-bottom: 1.0725rem`,
            },
          },
          `gatsby-remark-prismjs`,
          `gatsby-remark-copy-linked-files`,
          `gatsby-remark-smartypants`,
        ],
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-typescript`,
    {
      resolve: `gatsby-plugin-styled-components`,
      options: {
        displayName: process.env.NODE_ENV === "development",
      },
    },
  ],
}

export default config
