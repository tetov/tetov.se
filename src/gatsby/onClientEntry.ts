import type { GatsbyBrowser } from "gatsby"

const onClientEntry: GatsbyBrowser["onClientEntry"] = () => {
  // https://www.gatsbyjs.com/docs/browser-apis/#onClientEntry
  const polyfills: Promise<any>[] = []

  if (typeof fetch === "undefined") {
    polyfills.push(import("whatwg-fetch"))
  }

  // add your polyfills here

  return Promise.all(polyfills)
}

export default onClientEntry
