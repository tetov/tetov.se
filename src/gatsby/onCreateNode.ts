import type { GatsbyNode } from "gatsby"

const onCreateNode: GatsbyNode["onCreateNode"] = () => {
  // https://www.gatsbyjs.org/docs/node-apis/#onCreateNode
}

export default onCreateNode
