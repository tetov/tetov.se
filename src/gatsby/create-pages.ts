import { GatsbyNode } from "gatsby"
import path from "path"

import { IMarkdownRemarkNode } from "../helpers/types"

interface IQueryResult {
    allMarkdownRemark: {
        nodes: IMarkdownRemarkNode[]
    }
}

const createPages: GatsbyNode["createPages"] = async ({
  graphql,
  actions: { createPage },
  reporter,
}) => {
  // Get all markdown blog posts sorted by date
  const result = await graphql<IQueryResult>(
    `
      {
        allMarkdownRemark(
          sort: { fields: [frontmatter___date], order: ASC }
          limit: 1000
        ) {
          nodes {
            id
            fields {
              slug
              category
            }
          }
        }
      }
    `
  )

  if (result.errors || !result.data) {
    reporter.panicOnBuild(
      `There was an error loading your blog posts`,
      result.errors
    )
    return
  }

  const contentNodes = result.data.allMarkdownRemark.nodes

  // Create pages
  // `context` is available in the template as a prop and as a variable in GraphQL

  if (contentNodes.length > 0) {
    contentNodes.forEach((node) => {
      // const previousPostId = index === 0 ? null : posts[index - 1].id
      // const nextPostId = index === posts.length - 1 ? null : posts[index + 1].id
      const Component = path.resolve(
        `./src/components/templates/${node.fields.category || "pages"}.tsx`
      )

      createPage({
        path: node.fields.slug,
        component: Component,
        context: {
          id: node.id,
          //     previousPostId,
          //     nextPostId,
        },
      })
    })
  }
}

export default createPages
