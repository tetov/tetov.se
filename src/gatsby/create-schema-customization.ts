import { GatsbyNode } from "gatsby";

const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] =
  async ({ actions: { createTypes } }) => {
    // Explicitly define the siteMetadata {} object
    // This way those will always be defined even if removed from gatsby-config.js

    // Also explicitly define the Markdown frontmatter
    // This way the "MarkdownRemark" queries will return `null` even when no
    // blog posts are stored inside "content/blog" instead of returning an error
    createTypes(`
    type SiteMetadata {
      title: String!
      description: String
      siteURL: String!
      lang: String
    }
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
      fields: Fields
    }
    type Frontmatter {
      title: String
      description: String
      date: Date @dateformat
      lang: String
      hero: String
      weight: Int
    }
    type Fields {
      slug: String
      category: String
      heroImg: ImageSharp
    }
    type ContactData implements Node {
      hcard: String
      icon: String
      label: String!
      text: String
      url: String
      username: String
      icon: String
}
  `);
  };

export default createSchemaCustomization;
