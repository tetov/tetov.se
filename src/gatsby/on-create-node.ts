import { GatsbyNode } from "gatsby"
import path from "path"

import { parseNodePath } from "../helpers/node-path-operations"

const onCreateNode: GatsbyNode["onCreateNode"] = async ({
  actions: { createNodeField },
  node,
  getNode,
}) => {
  if (node.internal.type !== "MarkdownRemark") {
    return
  }

  const { dir, name } = parseNodePath(node, getNode)
  const dirArray = dir.split(path.sep)

  // Slug & Category

  // Check if node is part of a page bundle,
  // i.e. placed in its own dir and named index
  // category is two steps up from page bundle
  const isPageBundle = name === "index"

  const slug = isPageBundle ? dirArray[dirArray.length - 1] : name

  createNodeField({
    node: node,
    name: `slug`,
    value: slug,
  })

  const category = isPageBundle
    ? dirArray[dirArray.length - 2]
    : dirArray[dirArray.length - 1]

  createNodeField({
    node: node,
    name: `category`,
    value: category,
  })
}

export default onCreateNode
