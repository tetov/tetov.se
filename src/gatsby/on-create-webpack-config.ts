import { GatsbyNode } from "gatsby";
import path from "path";
const onCreateWebpackConfig: GatsbyNode["onCreateWebpackConfig"] = ({
  actions,
}) => {
  const srcDir = path.resolve(__dirname, "..");
  actions.setWebpackConfig({
    resolve: {
      alias: {
        components: path.resolve(srcDir, "components"),
        hooks: path.resolve(srcDir, "hooks"),
        images: path.resolve(srcDir, "images"),
        icons: path.resolve(srcDir, "icons"),
        types: path.resolve(srcDir, "types"),
        utils: path.resolve(srcDir, "utils"),
      },
    },
  });
};

export default onCreateWebpackConfig;
