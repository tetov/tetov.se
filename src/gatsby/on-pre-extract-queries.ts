import { GatsbyNode } from "gatsby"

import parseNodePath from "../helpers/parse-node-path"

import type { Node } from "gatsby"

type PreQueryImageSharp = Node & {
  parent: Node
}

type PreQueryMarkdownRemark = Node & {
  parent: Node
  fields: { isPageBundle: Boolean }
}

const onPreExtractQueries: GatsbyNode["onPreExtractQueries"] = async ({
  actions: { createNodeField },
  getNode,
  getNodesByType,
}) => {
  const allImgNodes = getNodesByType("ImageSharp") as PreQueryImageSharp[]
  const docNodes = getNodesByType("MarkdownRemark") as PreQueryMarkdownRemark[]

  const imgParsedPaths = {}
  allImgNodes.forEach((n) => (imgParsedPaths[n.id] = parseNodePath(n, getNode)))
  

  docNodes.forEach((docN) => {
    const { name: docName, dir: docDir } = parseNodePath(docN, getNode)
    console.log(`Name: ${docName} Dir: ${docDir}`)

    let _imgNodes = allImgNodes.filter(
      (imgN) => imgParsedPaths[imgN.id].dir === docDir
    )

    // Found no images
    if (_imgNodes.length < 1) return
    
    // https://stackoverflow.com/a/35092754/15398307
    _imgNodes.sort((a, b) =>
    imgParsedPaths[a.id].name.localeCompare(imgParsedPaths[b.id].name)
    )
    
    if (_imgNodes.length === 1 && (docN.fields.isPageBundle || imgParsedPaths[_imgNodes[0].id].name === docN.name)) {
      // Only use found image for hero if page bundle or named same as doc file
      if (!docN.fields.isPageBundle && imgParsedPaths[_imgNodes[0].id].name !== docN.name)
        return

      createNodeField({node: docN, name: "hero", value: _imgNodes[0]})
      return
    }

    const heroNodeIdx = _imgNodes.findIndex(n => imgParsedPaths[n.id].name === "hero") 
    
    // No image with name hero found, just use the first in (sorted) list
    if (heroNodeIdx === -1) {
      createNodeField({node: docN, name: "hero", value: _imgNodes[0]})
      createNodeField({node: docN, name: "imgs", value: _imgNodes.slice(1)})
      return
    }
    
    const heroNode = _imgNodes[heroNodeIdx]
    _imgNodes.splice(heroNodeIdx, 1)

    createNodeField({
      node: docN,
      name: `hero`,
      value: heroNode,
    })
  
    createNodeField({
      node: docN,
      name: `imgs`,
      value: _imgNodes,
    })
  })
}

export default onPreExtractQueries
