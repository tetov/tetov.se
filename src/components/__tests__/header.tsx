import * as Gatsby from "gatsby";
import React from "react";
import renderer from "react-test-renderer";
import Header from "src/components/header";

describe("Header", () => {
  it("renders correctly", () => {
    const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: `Nossnahoj Votet Notna`,
          description: `Mocks`,
          siteURL: `https://example.ai`,
          lang: `ent`,
          image: `/logo.png`,
        },
      },
    }));
    const tree = renderer
      .create(<Header subHeading={<span>Test</span>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
