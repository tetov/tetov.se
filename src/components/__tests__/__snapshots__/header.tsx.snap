// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="text-center md:justify-between pt-12 mb-16 md:mb-12"
>
  <a
    href="/"
  >
    <h1
      className="text-7xl md:text-8xl font-bold tracking-tighter leading-tight"
    >
      Nossnahoj Votet Notna
    </h1>
  </a>
   
  <h2
    className="mt-4 w-2/3 text-2xl md:text-4xl font-light inline-block leading-relaxed"
  >
    <span>
      Test
    </span>
  </h2>
</header>
`;
